nrf_802154/driver/include/nrf_802154.h

```diff
@@ -102,7 +102,7 @@ void nrf_802154_deinit(void);
  */
 extern void nrf_802154_custom_part_of_radio_init(void);
 
-#if !NRF_802154_INTERNAL_RADIO_IRQ_HANDLING
+#if !defined(NRF_802154_INTERNAL_RADIO_IRQ_HANDLING)
```

Comment:
Using `defined` and `#ifdef` instead of `#if` changes behavior of the driver.

Reply:
Doxygen doesn't seem to be able to process `#if !NRF_802154_INTERNAL_RADIO_IRQ_HANDLING`, not sure what is the best solution here.